examples/wmt/bleu.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""Approximate BLEU score.
```

Comment:
This is the score used for mlperf which tries to get close to sacrebleu - the reason I ported this over instead of just using the sacrebleu package is that it's actually easy to split up this implementation for very fast multi-host decoding and parallel bleu calculation.  I was planning to add that in a subsequent PR.

Reply:
(Drive by): SacreBLEU is the norm in machine translation, so that (or an approximation) is definitely what we want.
Question: where does this differ from sacrebleu?

Reply:
BLEU is originally defined by some perl script from 2003 (man, the nlp field and its metrics) - the field has since moved to sacreBLEU since it's a less crazy reference standard.  I think most differences come from subtle details of the string-tokenization that the bleu score uses in each case.